utils/kata-manager.sh

```diff
@@ -172,7 +186,7 @@ github_get_release_file_url()
 		-r '.[] |
 			select( (.tag_name == $version) or (.tag_name == "v" + $version) ) |
 			.assets[].browser_download_url' |\
-			grep "/${regex}$")
+			grep -E "/${regex}$")
```

Comment:
Hmm. I think that as one extra sanity check we should ensure that only 1 match is returned here:

```bash
local lines
lines=$(echo "$download_url" | wc -l | awk '{print $1}')

[ "$lines" -eq 1 ] || die "expected 1 download URL but found '$download_url'"
```

However, since I've already approved it, I'll let others comment on whether we should do that on this PR or whether we could do it on a follow-up PR though.

Reply:
I've raised #8364 to add this check later.

Reply:
Thank you for your guidance. It's valuable to include an extra sanity check for improved robustness.